tribe-ext-pdf-tickets/index.php

```diff
@@ -494,14 +507,31 @@ public function make_non_trailing_slash_the_canonical( $redirect_url, $requested
 
 
 	/**
-	 * Flush WordPress rewrite rules when this extension is activated.
-	 *
-	 * @link https://codex.wordpress.org/Function_Reference/register_post_type#Flushing_Rewrite_on_Activation
+	 * Ideally, we would only flush rewrite rules on plugin activation, but we
+	 * cannot use register_activation_hook() due to the timing of when
+	 * extensions load. Therefore, we flush rewrite rules on every visit to the
+	 * wp-admin Plugins screen (where we'd expect you to be if you just
+	 * activated a plugin)... only if our rewrite rule is not already in the
+	 * rewrite rules array.
 	 */
-	public function flush_rewrite_rules_on_activation() {
-		$this->add_pdf_file_rewrite_rules();
+	public function admin_flush_rewrite_rules_if_needed() {
+		global $pagenow;
+
+		if ( 'plugins.php' != $pagenow ) {
```

Comment:
No reason not to be strict here - let's switch to `!==`

Reply:
done